### PR TITLE
feature/checkbox-focus-blur

### LINKED ISSUE #1041: AXACheckbox: add onFocus and onBlur handlers
Add `onFocus` and `onBlur` handlers to `AXACheckbox` component.

These are needed by form frameworks which evaluate whether a field was touched and so whether to do validation, enable submit button, etc...

### PR DESCRIPTION
**Pull request by MKaHead**
fixes #1041

- Add blur and focus to checkbox
- Update readme and all boilerplate
- Improve stories and demos

### CODE REVIEW

File: src/components/10-atoms/checkbox/index.react.d.ts [hunk 1]
```diff
@@ -8,6 +8,8 @@ interface AXACheckboxProps {
   disabled?: boolean;
   error?: string;
   onChange?: (value: any) => void;
+  onFocus?: () => void;
```
Comment (MKaHead) [inline src/components/10-atoms/checkbox/index.react.d.ts]: @MarekLacoAXA do you want value: any here, too?

Comment (raphaellueckl) [reply src/components/10-atoms/checkbox/index.react.d.ts]: Why any? Can't it be `Function` or something like that?

File: src/components/10-atoms/checkbox/index.react.d.ts [hunk 2]
```diff
@@ -8,6 +8,8 @@ interface AXACheckboxProps {
   disabled?: boolean;
```
Comment (raphaellueckl) [inline src/components/10-atoms/checkbox/index.react.d.ts]: AFAIK, the lowercase types belong to javascript and the uppercase ones to typescript.

Comment (MarekLacoAXA) [reply src/components/10-atoms/checkbox/index.react.d.ts]: no, no.

**Pull request merged**